lnwallet/channel.go

```diff
@@ -2993,6 +3008,50 @@ func (lc *LightningChannel) evaluateHTLCView(view *HtlcView,
 			)
 			if rmvHeight == 0 {
 				switch {
+				// If this a noop add, then when we settle the
+				// HTLC, we actually credit the sender with the
+				// amount again, thus making it a noop. Noop
+				// HTLCs are only triggered by external software
+				// using the AuxComponents and only for channels
+				// that use the custom tapscript root.
+				case entry.EntryType == Settle &&
+					addEntry.EntryType == NoopAdd:
+
+					channel := lc.channelState
+					party := party
+
+					delta := btcutil.Amount(
+						balanceDeltas.GetForParty(
+							party,
+						),
+					)
+
+					// If the receiver has existing balance
+					// above dust then we go ahead with
+					// crediting the amount back to the
+					// sender. Otherwise we give the amount
+					// to the receiver. We do this because
+					// the receiver needs some above-dust
+					// balance to anchor the AuxBlob. We
+					// also pass in the so-far calculated
+					// delta for the party, as that's
+					// effectively part of their balance
+					// within this view computation.
+					if channel.BalanceAboveReserve(
```

Comment:
I think this should be an un-exported method on `lc` instead. Otherwise the delta parameter doesn't make a lot of sense IMO.